File: spec/javascript/test_nested_same_name2.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('NestedSameName2', 'src/nested_same_name2.bin', function(r) {
  assert.equal(r.version, 0x42);
  assert.equal(r.mainData.mainSize, 2);
  assert.equal(r.mainData.foo.data1.toString(), [0x11, 0x11, 0x11, 0x11].toString());
  assert.equal(r.dummy.dummySize, 3);
  assert.equal(r.dummy.foo.data2.toString(), [0x22, 0x22, 0x22, 0x22, 0x22, 0x22].toString());
});

File: spec/java/src/io/kaitai/struct/spec/TestNestedSameName2.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.NestedSameName2;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestNestedSameName2 extends CommonSpec {
    @Test
    public void testNestedSameName2() throws Exception {
        NestedSameName2 r = NestedSameName2.fromFile(SRC_DIR + "nested_same_name2.bin");

        assertEquals(r.version(), 0x42);
        assertEquals(r.mainData().mainSize(), 2);
        assertEquals(r.mainData().foo().data1(), new byte[] { 0x11, 0x11, 0x11, 0x11 });
        assertEquals(r.dummy().dummySize(), 3);
        assertEquals(r.dummy().foo().data2(), new byte[] { 0x22, 0x22, 0x22, 0x22, 0x22, 0x22 });
    }
}
